packages/asc-ui/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { ChevronLeft, ChevronRight } from '@datapunt/asc-assets'
+import PaginationStyle, {
+  Props,
+  PageNumber,
+  PreviousButton,
+  NextButton,
+} from './PaginationStyle'
+
+const Pagination: React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
+  page = 1,
+  collectionSize,
+  pageSize = 10,
+  onPageChange,
+  ...otherProps
+}) => {
+  const [currentPage, setCurrentPage] = React.useState<number>(page)
+  const totalPages = Math.ceil(collectionSize / pageSize)
+  const pageCluster = [currentPage]
```

Comment:
Should this not be put inside a `useEffect`?

Reply:
Tbh I'm not 100% clear on when it's ok to refrain from using useEffect in such scenarios. The computational effort here seems negligible to me. What's a good rationale /rule of thumb here? Can you have a look at what I converted it to now?